packages/rules/src/fileContents.ts

```diff
@@ -43,17 +43,21 @@ export const fileContents = {
     const generator = getGenerator(context, opts);
     const expectedContent = generator(context);
 
-    const actualContent = existsSync(fullPath) ? readFileSync(fullPath, "utf-8") : undefined;
+    const pathExists = existsSync(fullPath);
+    const actualContent = pathExists ? readFileSync(fullPath, "utf-8") : undefined;
 
     if (actualContent !== expectedContent) {
       context.addError({
         file: fullPath,
         message: "Expect file contents to match",
         longMessage: diff(expectedContent, actualContent, { expand: true }),
         fixer: () => {
-          if (expectedContent === undefined) {
+          if (expectedContent === undefined && pathExists) {
             unlinkSync(fullPath);
           } else {
+            if (!existsSync(path.dirname(fullPath))) {
+              mkdirSync(path.dirname(fullPath), { recursive: true });
```

Comment:
This is related to the bug we hit with earlier node versions. Should use the utility @maclockard wrote (in another pr?)

Reply:
what's the utility function called? i'm not sure what to look for

Reply:
It would need to be split out into its own function: https://github.com/monorepolint/monorepolint/blob/7cf85507b1b968a26b8e501bec25aefc12558c6e/packages/rules/src/__tests__/consistentDependencies.spec.ts#L71-L87 There's also possibly a better way to shim the `recursive: true` for `mkdirSync` if we think we want to us it elsewhere as well?